highlighters/main/main-highlighter.zsh

```diff
@@ -665,26 +665,16 @@ _zsh_highlight_highlighter_main_paint()
                  fi
                  style=reserved-word
                  ;;
-        $'\x7d') # right brace
-                 #
-                 # Parsing rule: # {
-                 #
-                 #     Additionally, `tt(})' is recognized in any position if neither the
-                 #     tt(IGNORE_BRACES) option nor the tt(IGNORE_CLOSE_BRACES) option is set."""
-                 if $right_brace_is_recognised_everywhere; then
+        *)       if false; then
+                 elif [[ $arg = '}' ]] && $right_brace_is_recognised_everywhere; then
```

Comment:
The reason for spelling it `$'\x7d'` in the incumbent code is to avoid $EDITOR's brace-matching matching a syntactical `{` with a single-quoted `}`. That helps $EDITOR flag mismatched braces.

Reply:
Ah, I would suggest that this is a bug in $EDITOR and it could use better syntax-highlighting :wink:. Does `'}'` work? I don't have non-alphanumeric ASCII memorized.

Reply:
Then how about adding a named constant `readonly RIGHT_CURLY`? Or a comment on the line having the magic number?

Reply:
I'd rather work around such EDITOR behaviour than force maintainers to deal with it.

Reply:
Fair enough; comment it is.